docs/src/coding-guidelines.md

```diff
@@ -0,0 +1,130 @@
+# Introduction
+
+This section outlines the coding standards used in the tt-mlir project. These guidelines are designed to enhance the readability and maintainability of our shared codebase. While these guidelines are not strict rules for every situation, they are essential for maintaining consistency across the repository.
+
+Our long-term aim is to have the entire codebase adhere to these conventions.
+
+Since our compiler is built on the LLVM MLIR framework, we strive to align closely with the LLVM coding style guidelines outlined here [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html).
+
+# Naming
+
+Clear and descriptive names are crucial for code readability and preventing bugs. It’s important to choose names that accurately reflect the semantics and purpose of the underlying entities, within reason. Avoid abbreviations unless they are widely recognized. Once you settle on a name, ensure consistent capitalization throughout the codebase to avoid confusion.
+
+General naming rule is to use *camel case* for most names (e.g., WorkaroundPass, isRankedTensor())
+
+1. Type Names
+    - Applies to classes, structs, enums, and typedefs.
+    - Should be *nouns* that describe the entity's purpose.
+    - Begin with an *upper-case* letter (e.g. TTNNOptimizerOptions, DecompositionPass)
+2. Variable Names
+    - Should be nouns, as they represent state.
+    - Use *camel case* and start with an *lower-case* letter (e.g. overrideInputLayout)
+3. Function Names
+    - Represent actions and should be verb phrases
+    - Use *camel case* and start with a *lower-case* letter (e.g. createTTNNOptimizer(), emitTTNNAsCpp())
+
+# Includes
+
```

Comment:
I'd suggest that all includes should be in "absolute form" (rooted at "ttmlir"), i.e. `#include "ttmlir/....something.h"` and never  `#include "something.h"`. This removes a number of potential problems with preprocessor, namespacing, etc.